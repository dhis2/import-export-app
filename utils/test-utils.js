import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
