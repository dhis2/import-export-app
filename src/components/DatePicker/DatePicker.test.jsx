import React from 'react'
import { render } from 'test-utils'

import { DatePicker } from '../index.js'

const props = {
    dataTest: 'date-picker',
    name: 'file',
    label: 'Start date',
    date: '2020-02-10',
    onChange: () => 1,
}

it('matches snapshot', () => {
    const { asFragment } = render(<DatePicker {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
