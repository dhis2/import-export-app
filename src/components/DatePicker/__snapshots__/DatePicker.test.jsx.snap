// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-1141087211 "
    data-test="date-picker"
  >
    <label
      class="jsx-2904627559 "
      data-test="date-picker-label"
      for="file"
    >
      <span
        class="jsx-2904627559"
      >
        Start date
      </span>
    </label>
    <div
      class="jsx-3651602664 "
      data-test="date-picker-content"
    >
      <div
        class="jsx-2908043323 "
        data-test="dhis2-uicore-box"
      >
        <div
          class="jsx-3353877153 jsx-31445346 input"
          data-test="dhis2-uicore-input"
        >
          <input
            class="jsx-3353877153 jsx-31445346 "
            id="file"
            name="file"
            type="date"
            value="2020-02-10"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
